doi_to_org.py

```diff
@@ -204,6 +209,20 @@ def tempfile_from_path(cls, path):
         shutil.copyfile(path, f.name)
         return f
 
+    @classmethod
+    def tempfile_from_key(cls, path, key):
+        if not os.path.isdir(path):
+            raise FileError('Not a valid path.')
+
+        for root, dirs, files in os.walk(path):
+            for f in files:
+                if os.path.splitext(f)[0] == key:
+                    temp_f = tempfile.NamedTemporaryFile()
+                    shutil.copyfile(os.path.join(root, f), temp_f.name)
+                    return temp_f
+
+        raise FileNotFoundError("Couldn't find file '%s.pdf' in '%s'" % (key, path))
+
```

Comment:
This is not always `key.pdf` now, as we relaxed this constraint. Maybe this message could do:
```python
raise FileNotFoundError("Could not find a file starting by '%s' in '%s'" % (key, path))
```